### PR TITLE
Congrats: remove unused failed purchase components

### PR DESCRIPTION
**Pull request by nightnei**
Related to https://github.com/Automattic/dotcom-forge/issues/6757

## Proposed Changes
After discussion here p9Jlb4-7Hx-p2#comment-11302 we decided to remove all "failed purchase" related code since we don't support this flow yet, so it's totally redundant to keep it.

## Testing Instructions
1) Visual review and tests should suffice
2) Additionally you can try to make a few different purchases and assert that everything still works well

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/header.jsx [hunk 1]
```diff
@@ -19,14 +17,12 @@ import {
 import getCheckoutUpgradeIntent from '../../../state/selectors/get-checkout-upgrade-intent';
 import './style.scss';
 
-export class CheckoutThankYouHeader extends PureComponent {
```
Comment (nightnei) [inline client/my-sites/checkout/checkout-thank-you/header.jsx:22]: We didn't use this export

File: client/my-sites/checkout/checkout-thank-you/index.tsx [hunk 2]
```diff
@@ -678,78 +673,6 @@ export class CheckoutThankYou extends Component<
...
-			upgradeIntent,
-			isSimplified,
-			displayMode,
-			receipt,
-		} = this.props;
-		const purchases = getPurchases( this.props );
-		const failedPurchases = getFailedPurchases( this.props );
-		const hasFailedPurchases = failedPurchases.length > 0;
-		const componentAndPrimaryPurchaseAndDomain = this.getComponentAndPrimaryPurchaseAndDomain();
-		const [ component, primaryPurchase ] = componentAndPrimaryPurchaseAndDomain;
```
Comment (nightnei) [inline client/my-sites/checkout/checkout-thank-you/index.tsx:715]: It's very weird line - `getComponentAndPrimaryPurchaseAndDomain` returns array with 1 element, no more. So all `primaryPurchase` code is unused and I removed it in this PR too.

File: client/my-sites/checkout/controller.jsx [hunk 3]
```diff
@@ -348,7 +348,6 @@ export function checkoutThankYou( context, next ) {
 				redirectTo={ context.query.redirect_to }
 				selectedFeature={ context.params.feature }
 				selectedSite={ selectedSite }
-				siteUnlaunchedBeforeUpgrade={ context.query.site_unlaunched_before_upgrade === 'true' }
```
Comment (nightnei) [inline client/my-sites/checkout/controller.jsx:351]: We don't use it anymore

File: client/my-sites/checkout/checkout-thank-you/header.jsx [hunk 4]
```diff
@@ -56,28 +47,7 @@ export class CheckoutThankYouHeader extends PureComponent {
...
-			"You will receive an email confirmation shortly for your purchase of {{strong}}%(productName)s{{/strong}}. What's next?",
-			{
-				args: {
-					productName: primaryPurchase.productName,
-				},
-				components: {
-					strong: <strong />,
-				},
-			}
-		);
```
Comment (chad1008) [inline client/my-sites/checkout/checkout-thank-you/header.jsx:80]: Curiosity, not feedback, to check my own "math":

Was this removal related to the failure page changes? Looking at the code it seems like it just wasn't being used as this header just wasn't being called for delayed domain transfer purchases, or even ecommerce purchases, before that bit of logic was [removed](https://github.com/Automattic/wp-calypso/pull/88572/files#diff-daac408bb350333e9fb8ca7c03233e75f03656e02a45c152e574227e6d253ee7L652-L675).

Again, not questioning the PR, just curious to confirm if I'm reading the history right 🙂

**Review by chad1008 — APPROVED** [review]
Nice cleanup! 🙂 
image

The changes look good to me, and I didn't notice anything unusual via smoke testing. :rocket: :ship:

File: client/my-sites/checkout/checkout-thank-you/header.jsx (hunk 4 above)
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/header.jsx:80]: Your "math" is completely accurate and correct 🙂 👍

**Pull request merged**